### PR TITLE
feat(LM): add Uniswap Liquidity mining script

### LINKED ISSUE #2258: Add Uniswap Liquidity mining script
**Why**

To accommodate additional LM programs that run on Uniswap (as opposed to balancer), an additional set of scripts are needed to be written.

**How**

Adapt the balancer scripts to accommodate the different dex.

**Resolution**

Scripts & tests are added.

**Difficulty Score [1-10]**

- 6: ~1 day

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

#2258

**Summary**

This PR adds an extra script to accommodate Liquidity mining on uniswap. This functionality differs slightly from how balancer's scripts work by enforcing that an LP is _also_ a token sponsor. This is done by commuting the redeemable number of synths that each LP token is worth and comparing this to the size of the token sponsor's position. The minimum between the redeemable tokens and the token sponsor's position size is used to enforce this rule.

Running this script is slightly different to the balancer scripts as the script requires to know: a) Uniswap pool address & EMP contract address. It uses a combination of these to find the "minimum" between the token sponsor size and LP position. to run the script execute the following with updated params.

```
node ./scripts/liquidity-mining/CalculateUniswapLPRewards.js --fromBlock 11356429 --toBlock 11379497 --poolAddress "0x25fb29d865c1356f9e95d621f21366d3a5db6bb0" --empAddress "0x516f595978D87B67401DaB7AfD8555c3d28a3Af4" --umaPerWeek 1000 --tokenName "ugas" --week 1 --network mainnet_mnemonic
```

**Issue(s)**

close #2258

### CODE REVIEW

File: packages/affiliates/liquidity-mining/CalculateBalancerLPRewards.js [hunk 1]
```diff
@@ -4,9 +4,8 @@
 // provided by for each liquidity provider to the single whitelisted pool.
 // -> For each snapshot block, calculate the $UMA rewards to be received by each liquidity provider based on the target weekly distribution.
 
-// Example usage from affiliates: node ./liquidity-mining/CalculateBalancerLPRewards.js --network mainnet_mnemonic \
-// --poolAddress="0x0099447ef539718bba3c4d4d4b4491d307eedc53" --fromDate = "2020-07-06" --toDate = "2020-07-13" --week \
-// = 1 --network mainnet_mnemonic
+// Example usage from core: node ./scripts/liquidity-mining/CalculateBalancerLPRewards.js --network mainnet_mnemonic --poolAddress="0x0099447ef539718bba3c4d4d4b4491d307eedc53" --fromBlock 10725993 --toBlock 10752010 --week=1
```
Comment (chrismaree) [inline packages/affiliates/liquidity-mining/CalculateBalancerLPRewards.js:7]: updated this comment for consistancy.

File: packages/affiliates/liquidity-mining/CalculateUniswapLPRewards.js [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+    shareHolder => uniswapPool.methods.balanceOf(shareHolder).call(undefined, blockNumber),
+    {
+      concurrency: 50 // Keep infura happy about the number of incoming requests.
+    }
+  );
+
+  // Also, get the position information for all shareholders.
+  const tokenShareHolderPositionResults = await Promise.map(
+    Object.keys(shareHolderPayout),
+    shareHolder => empContract.methods.positions(shareHolder).call(undefined, blockNumber),
```
Comment (daywiss) [inline packages/affiliates/liquidity-mining/CalculateUniswapLPRewards.js:194]: is there any issue with someone from shareholder payout not being in either uniswap pool or emp contract?

**Review by daywiss — APPROVED** [review]
nice work, looks good

File: packages/affiliates/liquidity-mining/CalculateUniswapLPRewards.js (hunk 2 above)
Comment (chrismaree) [reply packages/affiliates/liquidity-mining/CalculateUniswapLPRewards.js:194]: if they are not a uniswap LP then the Graph QL query will not return their address and they wont get iterated over in this step. We effectively start only by looking at uniswap LPs.

**Pull request merged**